src/status/wallet.nim

```diff
@@ -187,3 +187,15 @@ proc validateMnemonic*(self: WalletModel, mnemonic: string): string =
   result = status_wallet.validateMnemonic(mnemonic).parseJSON()["error"].getStr
 
 proc getAllCollectibles*(self: WalletModel, address: string): seq[Collectible] = getAllCollectibles(address)
+
+proc getGasPricePredictions*(self: WalletModel): GasPricePrediction =
+  try:
+    let url: string = fmt"https://etherchain.org/api/gasPriceOracle"
+    let client = newHttpClient()
+    client.headers = newHttpHeaders({ "Content-Type": "application/json" })
+    let response = client.request(url)
```

Comment:
Does this call freeze the UI? I ask because I saw below that you call it each time we open  the Send Modal. So it would be nice if the UI stayed reactive and we just update the gases once we get the return value.

Reply:
It does a tiny little bit prolly less then 400ms or so, but I totally agree that this should be using something like `spawnAndSend()`. Need to look into that.